darts/models/forecasting/regression_model.py

```diff
@@ -1017,8 +1039,144 @@ def _quantile_sampling(self, model_output: np.ndarray) -> np.ndarray:
         weights = (probs - left_q) / (right_q - left_q)
         inter = left_value + weights * (right_value - left_value)
 
+        # sampling has shape (n_series * n_samples, output_chunk_length, n_components)
         return inter.squeeze(-1)
 
+    def _poisson_sampling(self, model_output: np.ndarray) -> np.ndarray:
+        """
+        Model_output is of shape (n_series * n_samples, output_chunk_length, n_components)
+        """
+        return self._rng.poisson(lam=model_output).astype(float)
+
+    def _normal_sampling(self, model_output: np.ndarray) -> np.ndarray:
+        """Sampling method for CatBoost's [mean, variance] output.
+        model_output is of shape (n_components * output_chunk_length, n_samples, 2),
+        where the last 2 dimensions are mu and sigma.
+        """
+        n_entries, n_samples, n_params = model_output.shape
+        chunk_len = self.pred_dim
+
+        # treating each component separately
+        mu_sigma_list = [model_output[i, :, :] for i in range(n_entries)]
+
+        list_of_samples = [
+            self._rng.normal(
+                mu_sigma[:, 0],  # mean vector
+                mu_sigma[:, 1],  # diagonal covariance matrix
+            )
+            for mu_sigma in mu_sigma_list
+        ]
+
+        samples_transposed = np.array(list_of_samples).transpose()
+        samples_reshaped = samples_transposed.reshape(n_samples, chunk_len, -1)
+
+        return samples_reshaped
+
+    def _quantiles_params(self, model_output: np.ndarray) -> np.ndarray:
+        # model_outputs has shape (n_series * n_samples, output_chunk_length, n_components, n_quantiles),
+        # with n_samples = 1
+
+        # output shape (n_series * n_samples, output_chunk_length, n_components*n_quantiles)
+        shape = model_output.shape
+        if len(shape) <= 4:
+            return model_output
+        else:
+            # last dim : [comp_1_q_1, ..., comp_1_q_n, ..., comp_n_q_1, ..., comp_n_q_n]
+            return model_output.reshape(shape[0], shape[1], shape[2] * shape[3])
+
+    def _poisson_params(self, model_output: np.ndarray) -> np.ndarray:
+        # model_output has shape (n_series * n_samples, output_chunk_length, n_components), with n_samples = 1
```

Comment:
The output shape of predict and sample is somehow standardized, as `RegressionModel.predict()` expect something of the shape `(n_series * n_samples, output_chunk_length, n_components)` and then reshape it into `(n_series, output_chunk_length, n_components, n_samples)`.

I reworked the comments to make them less confusing, there were kind of notes to myself before that

It's the model_output shape for the gaussian likelihood (supported by CatBoost exclusively) that is slithly different from the two others, I can try to reshape it so that it's identical to the two others. Or maybe keep it for another PR, in which we could also move these LikelihoodMixin to the likelihood_models.py file. WDYT?